### ISSUE TITLE
Compiling errors with of 0.8.4 and 0.9.0RC2

### ISSUE THREAD

**Issue by livingbeams** [secondstory/ofxDSHapVideoPlayer#1]
Trying to compile the addon with OF 0.8.4 (VS2012) I get an error:
Error   9   error C2733: 'GetProcessInformation': no se permite una segunda vinculación C de la función sobrecargada  C:\development\of_v0.8.4_vs_release\libs\quicktime\include\Processes.h  195 1   ofxDSHapVideoPlayerExample
(in english: it is not allowed a second link of the overloaded function)

I've also tried to compilr with OF 0.9.0RC2 (VS2015) and I get these errors:

Severity    Code    Description Project File    Line
Error   C2259   'ofxDSHapVideoPlayer': cannot instantiate abstract class    ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0rc2_vs_release\apps\myapps\ofxdshapvideoplayerexample\src\ofapp.h   24
Error   C2259   'ofxDSHapVideoPlayer': cannot instantiate abstract class    ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0rc2_vs_release\apps\myapps\ofxdshapvideoplayerexample\src\ofapp.h   24
Warning D9030   '/Gm' is incompatible with multiprocessing; ignoring /MP switch ofxDSHapVideoPlayerExample  C:\development\of_v0.9.0RC2_vs_release\apps\myApps\ofxDSHapVideoPlayerExample\cl    1
Error   C2555   'ofxDSHapVideoPlayer::getPixels': overriding virtual function return type differs and is not covariant from 'ofBaseHasPixels_<unsigned char>::getPixels'    ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0rc2_vs_release\addons\ofxdshapvideoplayer\src\ofxdshapvideoplayer.h 33
Error   C2555   'ofxDSHapVideoPlayer::getPixels': overriding virtual function return type differs and is not covariant from 'ofBaseHasPixels_<unsigned char>::getPixels'    ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0rc2_vs_release\addons\ofxdshapvideoplayer\src\ofxdshapvideoplayer.h 33
Error   C2555   'ofxDSHapVideoPlayer::getPixels': overriding virtual function return type differs and is not covariant from 'ofBaseHasPixels_<unsigned char>::getPixels'    ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0rc2_vs_release\addons\ofxdshapvideoplayer\src\ofxdshapvideoplayer.h 33
Error   C2198   'PFNGLCOMPRESSEDTEXSUBIMAGE2DPROC': too few arguments for call  ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0rc2_vs_release\addons\ofxdshapvideoplayer\src\ofxdshapvideoplayer.cpp   1397
Error   C2039   'glTypeInternal': is not a member of 'ofTextureData'    ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0rc2_vs_release\addons\ofxdshapvideoplayer\src\ofxdshapvideoplayer.cpp   1397
Error   C2228   left of '.tryWait' must have class/struct/union ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0rc2_vs_release\addons\ofxdshapvideoplayer\src\ofxdshapvideoplayer.cpp   1364
Error   C2065   'cond': undeclared identifier   ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0rc2_vs_release\addons\ofxdshapvideoplayer\src\ofxdshapvideoplayer.cpp   1364
Error   C2039   'getCondition': is not a member of 'DirectShowHapVideo' ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0rc2_vs_release\addons\ofxdshapvideoplayer\src\ofxdshapvideoplayer.cpp   1363
Error   C2065   'cond': undeclared identifier   ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0rc2_vs_release\addons\ofxdshapvideoplayer\src\ofxdshapvideoplayer.cpp   1363
Error   C2065   'Condition': undeclared identifier  ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0rc2_vs_release\addons\ofxdshapvideoplayer\src\ofxdshapvideoplayer.cpp   1363
Error   C2039   'Condition': is not a member of 'Poco'  ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0rc2_vs_release\addons\ofxdshapvideoplayer\src\ofxdshapvideoplayer.cpp   1363
Error   C2039   'glTypeInternal': is not a member of 'ofTextureData'    ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0rc2_vs_release\addons\ofxdshapvideoplayer\src\ofxdshapvideoplayer.cpp   1218
Error   C2039   'glTypeInternal': is not a member of 'ofTextureData'    ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0rc2_vs_release\addons\ofxdshapvideoplayer\src\ofxdshapvideoplayer.cpp   1213
Error   C2039   'glTypeInternal': is not a member of 'ofTextureData'    ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0rc2_vs_release\addons\ofxdshapvideoplayer\src\ofxdshapvideoplayer.cpp   1209
Error   C2334   unexpected token(s) preceding '{'; skipping apparent function body  ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0rc2_vs_release\addons\ofxdshapvideoplayer\src\ofxdshapvideoplayer.cpp   1108
Error (active)      namespace "Poco" has no member "Condition"  ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0RC2_vs_release\addons\ofxDSHapVideoPlayer\src\ofxDSHapVideoPlayer.cpp   1108
Error   C4430   missing type specifier - int assumed. Note: C++ does not support default-int    ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0rc2_vs_release\addons\ofxdshapvideoplayer\src\ofxdshapvideoplayer.cpp   1108
Error   C2143   syntax error: missing ';' before '&'    ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0rc2_vs_release\addons\ofxdshapvideoplayer\src\ofxdshapvideoplayer.cpp   1108
Error   C2039   'Condition': is not a member of 'Poco'  ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0rc2_vs_release\addons\ofxdshapvideoplayer\src\ofxdshapvideoplayer.cpp   1108
Error (active)      namespace "Poco" has no member "Condition"  ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0RC2_vs_release\addons\ofxDSHapVideoPlayer\src\ofxDSHapVideoPlayer.cpp   1102
Error   C4430   missing type specifier - int assumed. Note: C++ does not support default-int    ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0rc2_vs_release\addons\ofxdshapvideoplayer\src\ofxdshapvideoplayer.cpp   1102
Error   C3646   'cond': unknown override specifier  ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0rc2_vs_release\addons\ofxdshapvideoplayer\src\ofxdshapvideoplayer.cpp   1102
Error   C2039   'Condition': is not a member of 'Poco'  ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0rc2_vs_release\addons\ofxdshapvideoplayer\src\ofxdshapvideoplayer.cpp   1102
Error   C2228   left of '.broadcast' must have class/struct/union   ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0rc2_vs_release\addons\ofxdshapvideoplayer\src\ofxdshapvideoplayer.cpp   226
Error   C2039   'cond': is not a member of 'DirectShowHapVideo' ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0rc2_vs_release\addons\ofxdshapvideoplayer\src\ofxdshapvideoplayer.cpp   226
Error (active)      class "ofTextureData" has no member "glTypeInternal"    ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0RC2_vs_release\addons\ofxDSHapVideoPlayer\src\ofxDSHapVideoPlayer.cpp   1209
Error (active)      class "ofTextureData" has no member "glTypeInternal"    ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0RC2_vs_release\addons\ofxDSHapVideoPlayer\src\ofxDSHapVideoPlayer.cpp   1213
Error (active)      class "ofTextureData" has no member "glTypeInternal"    ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0RC2_vs_release\addons\ofxDSHapVideoPlayer\src\ofxDSHapVideoPlayer.cpp   1218
Error (active)      namespace "Poco" has no member "Condition"  ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0RC2_vs_release\addons\ofxDSHapVideoPlayer\src\ofxDSHapVideoPlayer.cpp   1363
Error (active)      identifier "cond" is undefined  ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0RC2_vs_release\addons\ofxDSHapVideoPlayer\src\ofxDSHapVideoPlayer.cpp   1363
Error (active)      class "ofTextureData" has no member "glTypeInternal"    ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0RC2_vs_release\addons\ofxDSHapVideoPlayer\src\ofxDSHapVideoPlayer.cpp   1397
Error (active)      return type is not identical to nor covariant with return type "ofPixels_<unsigned char> &" of overridden virtual function "ofBaseHasPixels_<T>::getPixels [with T=unsigned char]"  ofxDSHapVideoPlayerExample  c:\development\of_v0.9.0RC2_vs_release\addons\ofxDSHapVideoPlayer\src\ofxDSHapVideoPlayer.h 33

**Comment by mantissa** [comment]
When compiling in 0.8.4, are you just compiling the example or are including the addon using your own application? I'll take a look at the addon in 0.9.0RC2 today.

**Comment by mantissa** [comment]
The issues you were seeing in 0.8.4 were conflicts between DirectShow and QuickTime. I've updated the source code to avoid this conflict (I'd just commented out the first function in QuickTime). The version of the addon for 0.8.4 is archived as a release here: https://github.com/secondstory/ofxDSHapVideoPlayer/releases. The code base has just been updated for 0.9.0. Let me know if you run into any other issues!
